elastic/kibana
# Make SameSite cookie's attribute configurable

## Summary

Closes: https://github.com/elastic/kibana/issues/60522
Adds support for `SameSite` attribute with patching Hapi Statehood library.
Another option would be to fork both `Statehood` & `hapi-auth-cookie` libs to extend `SameSite` validation with `None` value, but I decided that will require even more work.

@elastic/kibana-security How are we going to document this change? Should we mark the changes as `breaking`?
After Chrome resumes SameSite changes in July, customers using Kibana in **iframe** must:
- configure `SameSite` attribute in **kibana.yml**:
```yml
xpack.security.sameSiteCookies: None
```
- setup secure connection in **kibana.yml**, since [Cookies with SameSite=None must also specify Secure, meaning they require a secure context](https://web.dev/samesite-cookies-explained/#changes-to-the-default-behavior-without-samesite). Customers might need to tweak their setup to https://www.elastic.co/guide/en/kibana/current/configuring-tls.html#configuring-tls-browser-kib
```yml
server.ssl:
  enabled: true
  key: ....
  certificate: ...
  certificateAuthorities: ...
  clientAuthentication: ...
```
#### How to test locally
- Enable new logic for SameSite attribute
  - Chrome: https://www.chromium.org/updates/same-site/test-debug
  - Firefox:  [open about:config and set network.cookie.sameSite.laxByDefault](https://web.dev/samesite-cookies-explained/#changes-to-the-default-behavior-without-samesite)
- configure kibana.yml. you can generate certs or run Kibana with test ones:
```yml
xpack.security.sameSiteCookies: None
xpack.security.secureCookies: true
server.ssl:
  enabled: true
  key: packages/kbn-dev-utils/certs/kibana.key
  certificate: packages/kbn-dev-utils/certs/kibana.crt
  certificateAuthorities: packages/kbn-dev-utils/certs/ca.crt
  clientAuthentication: optional
```

@kobelb How deep should we backport this?

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers
